src/03-a-two-state-system-in-a-quantised-field.md

```diff
@@ -0,0 +1,602 @@
+
+two_state     =  A*sz                      # two state system energy
+bosons       =  omega*(a.dag()*a+0.5)      # bosons field energy
+interaction   = V*(a.dag() + a) * sx       # interaction energy
+
+H = two_state + bosons + interaction
+```
+
+Because we've got 10 possible states, the probability plots that we normally create are going to get a bit crowded. To make plots easier to understand, we'll create a function to label the simulation data according to $\pm$ notation that we've been using up to this point to describe the two state system.
+
+```python
+def states_to_df(states,times):
+    data = {}
+    for i in range(0,states[0].shape[0]):
+        which_mode = divmod(i,2)
+        if which_mode[1] == 0:
+            two_state = "+"
+        elif which_mode[1] == 1:
+            two_state = "-"
+        data[str(which_mode[0])+" , "+two_state] = np.zeros(len(times),dtype="complex128")
+    
+    for i, state in enumerate(states):
+        for j, psi in enumerate(state):
+            which_mode = divmod(j,2)
+            if which_mode[1] == 0:
+                two_state = "+"
+            elif which_mode[1] == 1:
+                two_state = "-"
+            data[str(which_mode[0])+" , "+two_state][i] = psi[0][0]
+
+    return pd.DataFrame(data=data, index=times)
+```
+
+The quintessential set-up for spontaneous emission is to have the field empty, i.e. no bosons $n=0$, and the two state system in it's "excited" (aka higher energy "+") state.
+
+Let's see what happens.
+
+```python
+psi0 = tensor(basis(max_bosons+1, 0), basis(2, 0))  # No bosons and two-state system is in the higher energy + state
+
+times = np.linspace(0.0, 70.0, 1000)      # simulation time
+
+result = sesolve(H, psi0, times)
+df_coupled =  states_to_df(result.states, times)
+```
+
+```python
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15,6))
+df_coupled.plot(title="Real part of amplitudes Re($\psi$)     (Fig 1)", ax=axes[0]);
+(df_coupled.abs()**2).plot(title="Probabilities $|\psi|^2$     (Fig 2))", ax=axes[1]);
+```
+
+It's taken us a while but we eventually arrived, as promised. We see in Fig 2 that even though we start with the field "empty" of bosons (blue line), after a while, the field is likely to have a single boson at the expense of the energy in the two state system which transitions to the lower state (red line) - the atom "spontaneously" "emits" a boson.
+
+
+You might also notice some other features:
+1. There is non-negligible chance to find the system in what appears to be a non-energy conserving state with 2 bosons and the two state system in the + state (purple line)
+2. The spontaneous emission doesn't appear to be irreversible (as we are normally taught)
+
+On point 1. To understand what's going on here, we need to take a deeper look into how the different bits of the Hamiltonian talk to each other and this will take us into the world of virtual states - it's a big enough topic to save for another tutorial.
```

Comment:
Not sure that the purple line | 2, + > is a virtual state since it can actually be measured

Reply:
I am conflicted here. From Wikipedia on [virtual states](https://en.wikipedia.org/wiki/Virtual_state) it sounds like |2,+> cannot be a virtual state because it's an eigenvector of H0 (i.e. without interaction). But, from [SJB blog](https://coldfusionblog.net/2014/08/20/the-lossy-part-of-the-lossy-spin-boson-model/) it does seem like  |2,+> is a virtual state because it's borrowing energy from "god". 

More discussion is required I think.

Reply:
From Wiki "no direct measurement of its energy is possible". Here I think that |2,+> is measurable, but that is because if you start at |0,+>, that is not an Eigenstate, but a mixture of eigenstates that evolves. So apparently you are violating the conservation of energy.

For sure more discussion required. I still don't get if the whole 'virtual state' stuff is only for perturbation theory approaches or what :(

Reply:
I also have a feeling that virtual stuff is something that comes out of perturbation theory.